main-app/app/scripts/MainController.js

```diff
@@ -1,91 +1,9 @@
-var app = angular.module("MainApp", [] );
+(function() {
+    'use strict';
+    angular.module("MainApp")
+        .controller("MainController", [ '$http', '$scope', '$location', 'NAndCModel', function($http, $scope, $location, NAndCModel) {
 
-app.controller("MainController", function($http, $scope) {
+            $scope.model = NAndCModel;
```

Comment:
NAndCModel is a bad choice of name - its' a noughts and crosses game, so gameModel  (with a provider name of "GameMode") is sufficient.